### DISCUSSION TITLE
CUPS to Zebra Printer Configuration

### DISCUSSION THREAD

**Discussion by singhpds** [OpenPrinting/cups#1034] [Q&A]
We are trying to send the Print Command from CUPS to Zebra ZT411 Label Network Printer (Wired/LAN). We have tried configuring the Printer on CUPS using Socket://PrinterIPAddress:9100 and Socket://PrinterIPAddress:6101 with generic ZPL driver however Zebra Printer is not printing anything and the CUPS logs shows, printing is successful. Please note that Telnet to printer IP Address is working. @zdohnal  - I have seen your post of a successful connection to Zebra printer from CUPS however not sure what we are missing here.[
![CUPS-Zebra](https://github.com/user-attachments/assets/c16a47ed-da02-46c5-818a-4e604558d0cf)
](url)

**Comment by michaelrsweet** [comment]
OK, so it should be port 9100 for talking to the printer's network port. You can also connect to the printer's embedded web server at "http://IP-ADDRESS" to verify the label and language configuration for the printer.

You should also make sure that the printer resolution is set to 203dpi (this is the default) and that the label type matches the labels you are using.

**Comment by singhpds** [comment]
Thank You Mike for your prompt response. Giving you additional information, We tried with port 9100 but it did not work too. We found an article to enable the PDF emulator on printer which we did however still print did not go through CUPS. I will verify again the label type. Interesting thing is data light keeps flashing while the LAN is connected and we have given any print commands.
